repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py

```diff
@@ -1,51 +1,85 @@
 
 
 class IPUWorkflow(Workflow):
+    """In-Place Upgrade workflow used by the leapp utility to process the in-place upgrade."""
+
     name = 'InplaceUpgrade'
-    tag = IPUWorkflowTag
+    tag = tags.IPUWorkflowTag
     short_name = 'ipu'
     configuration = IPUConfig
-    description = '''The IPU workflow takes care of an in-place upgrade (IPU) of RHEL 7 to RHEL 8.'''
+    description = """The IPU workflow takes care of an in-place upgrade (IPU) of RHEL 7 to RHEL 8."""
 
     class FactsCollectionPhase(Phase):
-        '''Get information (facts) about the system (e.g. installed packages, configuration, ...).
+        """
+        Get information (facts) about the system (e.g. installed packages, configuration, ...).
 
         No decision should be done in this phase. Scan the system to get information you need and provide
         it to other actors in the following phases.
-        '''
+        """
+
         name = 'FactsCollection'
-        filter = TagFilter(FactsPhaseTag)
+        filter = TagFilter(tags.FactsPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class ChecksPhase(Phase):
-        '''Check upgradability of the system, produce user question if needed and produce output for the report.
+        """
+        Check upgradability of the system, produce user question if needed and produce output for the report.
 
         Check whether it is possible to upgrade the system and detect potential risks. It is not expected to get
         additional information about the system in this phase, but rather work with data provided by the actors from
         the FactsCollection. When a potential risk is detected for upgrade, produce messages for the Reports phase.
-        '''
+        """
+
         name = 'Checks'
-        filter = TagFilter(ChecksPhaseTag)
+        filter = TagFilter(tags.ChecksPhaseTag)
+        policies = Policies(Policies.Errors.FailPhase,
+                            Policies.Retry.Phase)
+        flags = Flags()
+
+    class TargetFactsCollectionPhase(Phase):
+        """
+        Get information about target system. Analogy of FactsCollectionPhase for target system.
+
+        Here we can collect information what repositories are available on target system,
+        what is expected calculation of target transaction (what will be instaled, removed, ...
+        """
+
+        name = 'TargetFactsCollection'
+        filter = TagFilter(tags.TargetFactsPhaseTag)
+        policies = Policies(Policies.Errors.FailPhase,
+                            Policies.Retry.Phase)
+        flags = Flags()
+
+    class TargetCheckPhase(Phase):
+        """
+        Checks upgradability regarding the information gathered about the target system.
+
+        Check whether expected repositories and rpms are available, what rpms are planned
+        to install, remove, ...
+
+        IOW, checks related to rpm transaction mainly.
```

Comment:
This is kinda confusing. Can we just call it `TargetTransactionCheckPhase` or `TargetRPMCheckPhase` so that we're a little bit more explicit about what functionality we expect to be implemented there?

Reply:
I am thinking which of those two could be better. But both are better than the current one.

Reply:
My favorite is `TargetTransactionCheckPhase`, it seems most explicit about what is happening in that phase.

Reply:
So be it.